p2p/src/peer_manager/mod.rs

```diff
@@ -1466,6 +1473,14 @@ where
                 self.peerdb.unban(&address);
                 response_sender.send(Ok(()));
             }
+            PeerManagerEvent::Whitelist(address, response_sender) => {
+                self.peerdb.add_whitelisted_node(address);
```

Comment:
Shouldn't whitelisting be something in-memory? I might be wrong, but to me it sounds like whitelisting is something that can change per run.

Reply:
They are stored in memory only. I thought that it's a good place to put new set so that all address "groups" are managed by a single object. The fact the not everything is persisted might be confusing indeed. I can move it out to outer PeerManager